### PR TITLE
Feat/dataporten membership

### PR DESCRIPTION
**Pull request by sklirg**
## What kind of a pull request is this?

- QA / Code Review

## Code Checklist

- [x] The code follows dotkom code style 
- [x] The code passes the defined tests
- [x] I have added tests for the code I added
- [x] I have provided documentation for the code I added
- [x] The code is ready to be merged

## (Possible) Breaking Changes

- [x] The changes are backwards compatible

- [x] These changes requires changes to configuration in production 
    - [x] I have applied the required changes in production

## Description of changes

Adds a feature which asks the user to authorize through Dataporten, after which we fetch information about their groups (or in our case, courses). This information can be used to automatically approve or deny a membership application.

### CODE REVIEW

File: apps/dataporten/views.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    )
+
+    auth_req = client.construct_AuthorizationRequest(request_args=args)
+    login_url = auth_req.request(client.authorization_endpoint)
+
+    return redirect(login_url)
+
+
+@login_required()
+def study_callback(request):
```
Comment (hernil) [inline apps/dataporten/views.py]: Can this be split up somewhat?

File: apps/dataporten/study/tasks.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    study_name = study_group.get('displayName')
+    field_of_study = get_field_of_study(groups)
+
+    # Remove the years from bachelor if the user is a master student.
+    if study_year >= 4:
+        start_date_for_study = study_year - 3
+    else:
+        start_date_for_study = study_year
+
+    started_date = datetime(timezone.now().year - start_date_for_study, 7, 1)
```
Comment (hernil) [inline apps/dataporten/study/tasks.py]: Comment something about july first for clarity?

**Comment by hernil** [comment]
Haven't looked all to closely, nor tested, but from what I've seen this looks great!

File: apps/dataporten/study/tasks.py (hunk 2 above)
Comment (sklirg) [reply apps/dataporten/study/tasks.py]: Can do. :+1:

File: apps/dataporten/views.py (hunk 1 above)
Comment (sklirg) [reply apps/dataporten/views.py]: I tried, but I need access to the request for a lot of the messages, and both logging and messages end up spanning many lines. I'll look into it a bit more.

**Pull request merged**